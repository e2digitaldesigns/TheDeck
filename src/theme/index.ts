export * from "./styles/forms.styles";
export * from "./styles/scrollBars.styles";
