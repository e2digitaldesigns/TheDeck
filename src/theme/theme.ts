export const theme = {
  sizes: {
    application: {
      height: "785px",
      width: "1280px"
    },
    header: {
      height: "2rem",
      width: "100vw"
    },
    macroDeck: {
      sideBar: { width: "240px" },
      mainContent: { width: "1016px" }
    }
  },
  colors: {
    body: "black",
    border: "#111",
    accent1: "#f542e9",
    accent2: "#291697",
    accent3: "#404040",
    accent4: "#2b2b2b",
    accent5: "#1a1a1a",
    gridBg: "43, 45, 48,",
    font: "white",
    main: "#100727",
    mainBackground: "black",
    scrollBackground: "#100727",
    scrollFont: "#dedede",
    scrollHeader: "black",
    scrollHeaderFont: "white"
  }
};
