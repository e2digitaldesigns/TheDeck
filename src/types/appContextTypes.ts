export interface IntAppDataIconSelector {
  isVisible: boolean;
  icon: string;
}

export interface IntAppDataActive {
  profileId: string;
  pageId: string;
  buttonPadId: string;
  actionId: string;
}

export interface IntDropZones {
  actionList: boolean;
  buttonPads: boolean;
  sideBarProfiles: boolean;
  sideBarStyles: boolean;
  styleHeader: boolean;
}

export interface IntAppContextInterface {
  iconSelector: IntAppDataIconSelector;
  active: IntAppDataActive;
  dropZones: IntDropZones;
}

export interface IntAppContextStateInterface {
  appState: IntAppContextInterface;
  setAppState: React.Dispatch<React.SetStateAction<any>>;
}

export type IntAppData = IntAppContextStateInterface;

export enum AppStateActive {
  ProfileId = "profileId",
  PageId = "pageId",
  ButtonPadId = "buttonPadId",
  ActionId = "actionId"
}
