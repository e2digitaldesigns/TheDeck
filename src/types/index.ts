/* istanbul ignore file */
export * from "./actionType";
export * from "./aplicationTypes";
export * from "./appContextTypes";
export * from "./buttonPadGridTypes";
export * from "./dragAndDropTypes";
export * from "./globalContextType";
export * from "./globalType";
export * from "./ipcRenderTypes";
export * from "./obsTypes";
