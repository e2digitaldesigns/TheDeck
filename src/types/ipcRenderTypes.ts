export enum IpcRendererTypes {
  database = "database",
  databaseReturn = "database:return",
  header = "header",
  importSettings = "importSettings",
  importSettingsReturn = "importSettings:return",
  mdPlaySound = "MD:playSound"
}

export enum IpcRendererTypesAction {
  close = "close",
  fsToggle = "fsToggle",
  minimize = "minimize",
  loadAppData = "loadAppData",
  saveAppData = "saveAppData"
}
