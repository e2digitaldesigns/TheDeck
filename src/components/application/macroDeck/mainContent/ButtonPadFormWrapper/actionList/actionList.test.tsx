import React from "react";
import "@testing-library/jest-dom";
import { cleanup, fireEvent, render } from "@testing-library/react";
import _cloneDeep from "lodash/cloneDeep";
import ActionList from "./actionList";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../../../../theme/theme";

import initAppState from "../../../../../../__mock_data__/appState.json";
import initGlobalState from "../../../../../../__mock_data__/globalState.json";
import { IntAppData, IntGlobalData } from "../../../../../../types";

const mockAppState: IntAppData = {
  appState: _cloneDeep(initAppState),
  setAppState: jest.fn(appState => appState)
};

const mockGlobalState: IntGlobalData = {
  state: _cloneDeep(initGlobalState),
  setState: jest.fn(state => state)
};

const mockHooks = {
  useAppData: mockAppState,
  useGlobalData: mockGlobalState,
  useActions: {
    activateAction: jest.fn(),
    createAction: jest.fn(),
    getActions: jest.fn(),
    deleteAction: jest.fn()
  },
  actionItem: mockGlobalState.state.actions[1]
};

jest.mock("../../../../../../hooks", () => ({
  useAppData: () => mockHooks.useAppData,
  useGlobalData: () => mockHooks.useGlobalData,
  useActions: () => ({
    activateAction: mockHooks.useActions.activateAction,
    createAction: mockHooks.useActions.createAction,
    getActions: mockHooks.useActions.getActions.mockImplementation(() => [
      mockHooks.actionItem
    ]),
    deleteAction: mockHooks.useActions.deleteAction
  }),
  usePage: () => jest.fn(),

  useProfile: () => ({ readProfile: jest.fn() })
}));

const testSetup = () => {
  return render(
    <ThemeProvider theme={theme}>
      <ActionList />
    </ThemeProvider>
  );
};

let wrapper: any = null;

beforeEach(() => {
  // wrapper = testSetup();
});

afterEach(() => {
  wrapper = null;
  mockAppState.appState = _cloneDeep(initAppState);
  jest.resetAllMocks();
  jest.clearAllMocks();
  cleanup;
});

describe("<ActionList/>", () => {
  it("Should render without errors", () => {
    wrapper = testSetup();
    expect(wrapper).toMatchSnapshot();
  });

  it("Should call getActions", () => {
    wrapper = testSetup();
    expect(mockHooks.useActions.getActions).toHaveBeenCalled();
  });

  it("Should call activateAction", () => {
    wrapper = testSetup();
    const component = wrapper.queryByTestId(
      "action-list__action-item-activate"
    );
    expect(component).toBeTruthy();

    fireEvent.click(component);
    expect(mockHooks.useActions.activateAction).toHaveBeenCalledTimes(1);
    expect(mockHooks.useActions.activateAction).toHaveBeenCalledWith(
      mockHooks.actionItem._id
    );
  });

  it("Should call createAction", () => {
    mockAppState.appState.active.buttonPadId = mockHooks.actionItem.buttonPadId;
    wrapper = testSetup();

    const component = wrapper.getByTestId("action-list__action-item-new");
    expect(component).toBeTruthy();

    fireEvent.click(component);
    expect(mockHooks.useActions.createAction).toHaveBeenCalledTimes(1);
  });

  it("Should call deleteAction", () => {
    wrapper = testSetup();

    const component = wrapper.getByTestId("action-list__action-item-delete");
    expect(component).toBeTruthy();

    fireEvent.click(component);
    expect(mockHooks.useActions.deleteAction).toHaveBeenCalledTimes(1);
    expect(mockHooks.useActions.deleteAction).toHaveBeenCalledWith(
      mockHooks.actionItem._id
    );
  });

  it("Should Not call createAction", () => {
    wrapper = testSetup();
    const component = wrapper.getByTestId("action-list__action-item-new");
    expect(component).toBeTruthy();
    expect(component).toHaveAttribute("disabled");

    fireEvent.click(component);
    expect(mockHooks.useActions.createAction).toHaveBeenCalledTimes(0);
  });
});
