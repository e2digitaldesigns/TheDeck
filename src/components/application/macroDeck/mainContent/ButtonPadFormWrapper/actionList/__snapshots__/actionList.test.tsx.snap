// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionList/> Should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jRQBWg fxxBMb"
      >
        <div
          class="sc-eCImPb sc-gKclnd kFGBBw dkZvxI"
        >
          <ul>
            <li
              class="sc-iCfMLu bbPBDp"
              data-testid="action-list__action-item-activate"
            >
              <div
                class="sc-furwcr gOsykF"
              >
                <div>
                  obs
                </div>
                <div>
                   | 
                </div>
                <div
                  data-testid="sub-action-parser__text"
                >
                  OBS Stream Toggle
                </div>
              </div>
              <button
                class="sc-hKwDye sc-pVTFL PLLBx jwZTjl"
                data-testid="action-list__action-item-delete"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </button>
            </li>
          </ul>
        </div>
      </div>
      <button
        class="sc-hKwDye sc-jrQzAO PLLBx udMZG"
        data-testid="action-list__action-item-new"
        disabled=""
      >
        New Action
      </button>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jRQBWg fxxBMb"
    >
      <div
        class="sc-eCImPb sc-gKclnd kFGBBw dkZvxI"
      >
        <ul>
          <li
            class="sc-iCfMLu bbPBDp"
            data-testid="action-list__action-item-activate"
          >
            <div
              class="sc-furwcr gOsykF"
            >
              <div>
                obs
              </div>
              <div>
                 | 
              </div>
              <div
                data-testid="sub-action-parser__text"
              >
                OBS Stream Toggle
              </div>
            </div>
            <button
              class="sc-hKwDye sc-pVTFL PLLBx jwZTjl"
              data-testid="action-list__action-item-delete"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </div>
    </div>
    <button
      class="sc-hKwDye sc-jrQzAO PLLBx udMZG"
      data-testid="action-list__action-item-new"
      disabled=""
    >
      New Action
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
